src/vm/wren_vm.c

```diff
@@ -1460,9 +1461,10 @@ WrenInterpretResult wrenCall(WrenVM* vm, WrenHandle* method)
 
   // Discard any extra temporary slots. We take for granted that the stub
   // function has exactly one slot for each argument.
-  vm->fiber->stackTop = &vm->fiber->stack[closure->fn->maxSlots];
+  const int closureNumArgs = closure->fn->arity + 1;
+  vm->fiber->stackTop = &vm->fiber->stack[closureNumArgs];
   
-  wrenCallFunction(vm, vm->fiber, closure, 0);
+  wrenCallFunction(vm, vm->fiber, closure, closureNumArgs);
```

Comment:
this code might not behave as expected? The closure being called here is the stub function created by the call handle, which only has bytecode to forward the call to the receiver. The closure itself has 0 args marked here specifically to avoid interacting with the stack afaict

Reply:
The old behavior was ONLY correct when passing the correct number of arguments using broken logic. There was a bogus relation between what the last `wrenCallFunction` argument was intended to be and what it was really used for. With that change, the argument is now the number of arguments that are on the stack to perform the execution call. Because of that it is not 0 anymore, since obviously a closure can always have any number arguments.
Because a closure has to accept its arity or more arguments, as per `Fn.call`, there MUST be a stack interaction if there is more arguments than required.